### ISSUE TITLE
Quantiation TODO

### ISSUE THREAD

**Issue by Luonic** [Luonic/semi-supervised_semantic_segmentation#1]
- [ ] Replace addition with:
```
self.skip_add = nn.quantized.FloatFunctional()
def forward(self, x):
    return self.skip_add.add(x, block(x))
```
- [ ] Insert QuantStub and DeQuantStub at the beginning and end of the network.
- [ ] Fuse model with:
```
def fuse_model(self):
        for m in self.modules():
            if type(m) == ConvBNReLU:
                torch.quantization.fuse_modules(m, ['0', '1', '2'], inplace=True)
            if type(m) == InvertedResidual:
                for idx in range(len(m.conv)):
                    if type(m.conv[idx]) == nn.Conv2d:
                        torch.quantization.fuse_modules(m.conv, [str(idx), str(idx + 1)], inplace=True)
float_model.eval()

# Fuses modules
float_model.fuse_model()
```
- [ ] Quantization-aware training:
```
qat_model.qconfig = torch.quantization.get_default_qat_qconfig('fbgemm') # or 'qnnpack' for ARM
torch.quantization.prepare_qat(qat_model, inplace=True)
```
Training a quantized model with high accuracy requires accurate modeling of numerics at inference. For quantization aware training, therefore, we modify the training loop by:
- Switch batch norm to use running mean and variance towards the end of training to better match inference numerics.
- We also freeze the quantizer parameters (scale and zero-point) and fine tune the weights.
```
for nepoch in range(8):
    train_one_epoch(qat_model, criterion, optimizer, data_loader, torch.device('cpu'), num_train_batches)
    if nepoch > 3:
        # Freeze quantizer parameters
        qat_model.apply(torch.quantization.disable_observer)
    if nepoch > 2:
        # Freeze batch norm mean and variance estimates
        qat_model.apply(torch.nn.intrinsic.qat.freeze_bn_stats)
```
quantized_model = torch.quantization.convert(qat_model.eval(), inplace=False)
